xdsl/dialects/mpi.py

```diff
@@ -128,16 +139,17 @@ class Send(MPIBaseOp):
 
     name = 'mpi.send'
 
-    buffer: Annotated[Operand, MemRefType[AnyNumericType]]
+    buffer: Annotated[Operand, AnyAttr()]
+    count: Annotated[Operand, i32]
+    datatype: Annotated[Operand, DataType]
     dest: Annotated[Operand, i32]
-
-    tag: OpAttr[IntegerAttr[Annotated[IntegerType, i32]]]
+    tag: Annotated[Operand, i32]
 
     @classmethod
-    def get(cls, buff: SSAValue | Operation, dest: SSAValue | Operation,
-            tag: int) -> Send:
-        return cls.build(operands=[buff, dest],
-                         attributes=_build_attr_dict_with_optional_tag(tag),
+    def get(cls, buffer: SSAValue | Operation, count: SSAValue | Operation,
+            datatype: SSAValue | Operation, dest: SSAValue | Operation,
+            tag: SSAValue | Operation) -> Send:
+        return cls.build(operands=[buffer, count, datatype, dest, tag],
```

Comment:
The `cls` thing again.

Reply:
Okay, this is pretty much everywhere. Maybe make this an issue and do it in a different PR?